### PR TITLE
macOS support for run_protoc.sh

### PR DESCRIPTION
**Pull request by mattes-bru**
I stumbled over an incompatibility with the macOS version of sed. I got this solved by doing this little hack using the homebrew gnu-sed package. There might be better solutions but I am neither a bash nor an regex expert but I think I'd better come with a solution than a problem. 😄

### CODE REVIEW

File: other/tools/run_protoc.sh [hunk 1]
```diff
@@ -24,7 +24,11 @@ command -v protoc-gen-mavsdk > /dev/null || {
 }
 
 function snake_case_to_camel_case {
-    echo $1 | sed -r 's/(^|_)([a-z])/\U\2/g'
+    if [[ "$OSTYPE" == "darwin"* ]]; then
```
Comment (julianoes) [inline other/tools/run_protoc.sh:27]: Where does `$OSTYPE` come from?

File: other/tools/run_protoc.sh [hunk 2]
```diff
@@ -24,7 +24,11 @@ command -v protoc-gen-mavsdk > /dev/null || {
...
+        echo $1 | gsed -r 's/(^|_)([a-z])/\U\2/g'
```
Comment (julianoes) [inline other/tools/run_protoc.sh:28]: Does `gsed` exist on every Mac?

**Review by julianoes — COMMENTED** [review]
Sorry, I was meaning to cherry-pick my fix that I did here:
https://github.com/mavlink/MAVSDK/commit/f158136941c3b92843058fdf780a94179bd6405c

Now we just need to decide which one we prefer.

File: other/tools/run_protoc.sh (hunk 1 above)
Comment (mattes-bru) [reply other/tools/run_protoc.sh:27]: It is from one of the first search hits on "how to determine the os type in bash scripts" 😀 

It seems to be defined by default.

File: other/tools/run_protoc.sh (hunk 2 above)
Comment (mattes-bru) [reply other/tools/run_protoc.sh:28]: No, it needs to be installed bye 'brew install gnu-sed'

**Comment by mattes-bru** [comment]
Based on your comments and my responses, I would prefer yours, because it seems to be OS independent without any additional packages.

**Comment by julianoes** [comment]
> Based on your comments and my responses, I would prefer yours, because it seems to be OS independent without any additional packages.

Me too. I was just trying to cherry-pick it but now realized it is already applied, however, that's in the C++ repo, not the Python one! Would you mind making a PR and using the awk magic?

**Comment by mattes-bru** [comment]
@julianoes I pushed your suggestion to my branch. This should be enough to update this PR, otherwise let me know. I also tested it on my machine and it worked as expected.

**Review by julianoes — APPROVED** [review]

**Comment by julianoes** [comment]
Great, thank you!

**Review by JonasVautherin — APPROVED** [review]

**Pull request merged**